lib/feed.js

```diff
@@ -0,0 +1,55 @@
+const data = require('../out/data.json')
+const feedFinder = require('find-rss')
+const fs = require('fs')
+const feedParser = require('feed-read-parser')
+
+const errs = []
+let articles = []
+
+function getFeeds(feedUrls, i, cb) {
+  let feedUrl = feedUrls[i]
+  if (!feedUrl) return cb(errs, articles)
+  feedParser.get(feedUrl, function(err, _articles) {
+    if (err) errs.push(err.message + ' ' + feedUrl)
+    else {
+      articles = articles.concat(_articles)
+    }
+    getFeeds(feedUrls, i + 1, cb)
+  })
+}
+
+async function getFeedLinks(data) {
+  const linkPromises = data.map(org => {
+    const { blog_url } = org
+    if (blog_url) {
+      return feedFinder(blog_url)
+        .catch(err => console.error(err))
+        .then(res => {
+          if (res.length > 0) {
+            return res[0].url
+          }
+        })
+    }
+  })
+  return await Promise.all(linkPromises)
+}
+
+function wordInString(s, word) {
+  return new RegExp('\\b' + word + '\\b', 'i').test(s)
+}
+
+;(async () => {
+  const feedLinks = (await getFeedLinks(data)).filter(link => link)
+  getFeeds(feedLinks, 0, (errs, articles) => {
+    if (errs.length > 0) {
+      errs.foreach(err => console.error(err))
+    }
+    const feeds = articles.filter(
+      feed =>
+        (new Date(feed.published) > new Date('November 27, 2017')) |
+        wordInString(feed.content, 'Google Code-In')
+    )
+    feeds.sort((a, b) => new Date(b.published) - new Date(a.published))
+    fs.writeFileSync(`${__dirname}/../out/feeds.json`, JSON.stringify(feeds))
```

Comment:
this is creating *one* new feed, not "feeds".  And it contains a specific type of articles, so lets call it "blog_planet.json".

After this PR is merged, please create a new issue to emit the feed as an RSS or ATOM file, so it can be read using a RSS reader.  And that file should be called  `blog_planet.rss` or `blog_planet.atom`.

Reply:
Also rename `lib/feed.js` to `lib/planet.js` .

"gci-leaders" may also start publishing its own RSS feed, including any updates to the leaderboards ;-)